docs/syntax/Cadence.syntax

```diff
@@ -0,0 +1,1482 @@
+        ":" expression
+    ;
+
+statements:
+        *eosStatement
+    ;
+
+eosStatement:
+        eos
+    |   statement
+    ;
+
+eos:
+        semicolon
+    |   newLine
+    ;
+
+newLine:
+        /
+    ;
+
+statement:
+        returnStatement
+    |   breakStatement
+    |   continueStatement
+    |   ifStatement
+    |   whileStatement
+    |   forStatement
+    |   switchStatement
+    |   emitStatement
+    |   declaration
+    |   assignment
+    |   swap
+    |   expression
+    ;
+
+returnStatement:
+        returnStatementExpression
+    |   returnStatementEmpty
+    ;
+
+returnStatementExpression:
+        returnKeyword expression
+    ;
+
+returnStatementEmpty:
+        returnKeyword
+    ;
+
+breakStatement:
+        breakKeyword
+    ;
+
+continueStatement:
+        continueKeyword
+    ;
+
+ifStatement:
+        ifKeyword ifStatementCondition block *1ifStatementElse
+    ;
+
+ifStatementCondition:
+        expression
+    |   variableDeclaration
+    ;
+
+ifStatementElse:
+        elseKeyword ifStatementElseElement
+    ;
+
+ifStatementElseElement:
+        ifStatement
+    |   block
+    ;
+
+whileStatement:
+        whileKeyword expression block
+    ;
+
+forStatement:
+        forKeyword *1forStatementIndex nonReservedIdentifier inKeyword expression block
+    ;
+
+forStatementIndex:
+        nonReservedIdentifier ","
+    ;
+
+switchStatement:
+        switchKeyword expression "{" *switchCase "}"
+    ;
+
+switchCase:
+        switchCaseExpression
+    |   switchCaseDefault
+    ;
+
+switchCaseExpression:
+        caseKeyword expression ":" statements
+    ;
+
+switchCaseDefault:
+        defaultKeyword ":" statements
+    ;
+
+emitStatement:
+        emitKeyword 1*identifier( "." ) *1invocationTemplateParameters "(" *1argumentList ")"
+    ;
+
+variableDeclaration:
+        letVariableDeclaration
+    |   varVariableDeclaration
+    ;
+
+letVariableDeclaration:
+        access letKeyword nonReservedIdentifier *1variableDeclarationTypeAnnotation transfer expression *1variableDeclarationTrailingTransferExpression
+    ;
+
+varVariableDeclaration:
+        access varKeyword nonReservedIdentifier *1variableDeclarationTypeAnnotation transfer expression *1variableDeclarationTrailingTransferExpression
+    ;
+
+variableDeclarationTypeAnnotation:
+        ":" typeAnnotation
+    ;
+
+variableDeclarationTrailingTransferExpression:
+        transfer expression
+    ;
+
+assignment:
+        assignmentTarget transfer assignmentValue
+    ;
+
+assignmentTarget:
+        expression
+    ;
+
+assignmentValue:
+        expression
+    ;
+
+swap:
+        swapExpressionLeft "<->" swapExpressionRight
+    ;
+
+swapExpressionLeft:
+        expression
+    ;
+
+swapExpressionRight:
+        expression
+    ;
+
+transfer:
+        transferEqual
+    |   transferMoveForced
+    |   transferMove
+    ;
+
+transferEqual:
+        "="
+    ;
+
+transferMoveForced:
+        moveForced
+    ;
+
+transferMove:
+        move
+    ;
+
+expression:
```

Comment:
The precedence numbers are an implementation detail of the current parser implementation, maybe remove them from the document as they are not necessary and probably just confusing. The rules are already structured in a way where they encode the precedence rules.

Reply:
That was the intention. They were there only to help define the grammar. I forgot to remove them.